procrustes/psdp.py

```diff
@@ -34,10 +34,90 @@
 __all__ = [
     "psdp_woodgate",
     "psdp_peng",
-    "psdp_opt"
+    "psdp_opt",
+    "psdp_projgrad"
```

Comment:
```python
"psdp_projgrad",
```
It's a good practice to have a comma to make the Git history clean.

Reply:
In the end we may wish to split off some helper functions. The key criterion is that helper functions that are generically useful (not tired to pdsp) can be split off, provided we can make them "generic" enough to be useful outside the pdsp context. Otherwise better to keep encapsulated with the rest of the pdsp code. 

It's really a matter of style. In the bad old days of programming, people deal with *very* long files. Nowadays people prefer shorter files, but if the number of files gets too large, it can be difficult to find what you are looking for in the code. Ideal, I think, is to hopefully make it so that when you need to find something, you can guess which file it will be in with two-three guesses, which suggests relatively few files, but not too few.